src/components/implementation/no_interface/llbooter/llbooter.c

```diff
@@ -348,14 +427,50 @@ void
 init_done(int parallel_init, init_main_t main_type)
 {
 	compid_t client = (compid_t)cos_inv_token();
-	struct crt_comp *c;
+	struct crt_comp    *c;
+	struct crt_chkpt   *chkpt;
+	struct crt_comp    *new_comp  = boot_comp_get(crt_ncomp() + 1);
+	thdcap_t 		thdcap;
+	int 			ret;
+	char			*name;
 
 	assert(client > 0 && client <= MAX_NUM_COMPS);
 	c = boot_comp_get(client);
 
 	crt_compinit_done(c, parallel_init, main_type);
 
+#ifdef ENABLE_CHKPT
+	if (c->id == cos_compid()) {
```

Comment:
Can we throw all of this `ENABLE_CHKPT` block in a separate function? It isn't great for it to make the main path more complex, and it would be good to have most of the checkpoint logic together.